### ISSUE TITLE
Object literal may only specify known properties, and 'extend' does not exist in type 'Theme'.

### ISSUE THREAD

**Issue by michealroberts** [unocss/unocss#2806]
### UnoCSS version

0.53.1

### Describe the bug

Essentially, when I attempt to extend the theme, I get the following TS error:

```ts
Object literal may only specify known properties, and 'extend' does not exist in type 'Theme'.
```

<img width="930" alt="Screenshot 2023-06-28 at 17 18 28" src="https://github.com/unocss/unocss/assets/84131395/66dc2165-5c0d-43bc-8eb3-d26fe7cfec5b">

As an example, from the UnoCSS playground: https://uno.antfu.me/play/?html=DwEwlgbgBAxgNgQwM5ILwHIAuBTAHpgWhmwDscAndAPiigChbgAHKAIwHMD3yFxTMMmAPYFyUAGbkhAWwKs4AV2wEALAAY16WIhQYkmMDADWATyjCmBTVQSsYIbOPYALMACsjwAPRMqDKMzayGjoAO4EAOxqTLhQCCRg0gg4BKFg7Oxw2FocGFJIyurWaRlZ3r7%2Bgc4EAIwAzGoQzmxC5A7ktTZ25TT03uAQVEA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmcAvnOlBCHAOQCuWEAxgM6u0BQnqAHpLBQYAhvQA28NJhz5CwIgAoEnOHBgALVCFQAuRCtVw%2BMVFmR7lhwwGtUAT0rDtrCwauqA7vKJjd%2B9%2B60AAwApAA0cACMQaG0FmpQwlis6NAgerRUMMImCgC0AMxoRACUtGRhbgF0AKyx8TCJyalQ6XRZOagKBUHFZRVVVqSDAwFJoDnABK7VcF5EPn6084tRrEbCrKh52HkQ9PDY0sAmtJUBw0MGl3CcpCVAA&options=N4IgzgLgTglgxhEAuAZgQwDZgKYF8g

This example throws a TS error as above.

### Reproduction

```ts
export default defineConfig({
  theme: {
    extend: {
        // Some extended theme properties here ...
    }
  }
})
```

### System Info

Vite 4.3.9
Vue 3.3.4
Typescript 5.1.3
UnoCSS 0.53.1

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't [already an issue](https://github.com/unocss/unocss/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/unocss/unocss/discussions) or join our [Discord Chat Server](https://chat.antfu.me).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by zyyv** [comment]
Refer to https://github.com/unocss/unocss/discussions/2012.

**Comment by michealroberts** [comment]
I feel this needs to be re-opened, because the ticket referenced is a different object in the config. This is about extending the theme … ?

**Comment by zyyv** [comment]
> I feel this needs to be re-opened, because the ticket referenced is a different object in the config. This is about extending the theme … ?

Hi, please use `extendTheme` extend the theme . [Playground](https://unocss.dev/play/?html=DwEwlgbgBA7gTgQwA5IKZwHwCgpVJKAF1QA9CBaYgZ0IwFUA7AewGEBlN4AenAm297YgA&config=JYWwDg9gTgLgBAbwFBzgEwKYDNgDsMDCEuOA5gDQpxhQYDOGMAgjDFMAEYCuMwWAnpVQ16jAKq4IlAL5wsUCCDgByLpIDGdOsqRIMAD0ix02AIZcANvEw58REsFIAKZKjoALaDHU86ALkQqVGUAdyhTMDAMKGUA0IBaAEYABmSANxC4dyTUtPc5CwM4YBgMEDp49QxcUqg4ACsuOl4BSuranVRpITgDUtw0ABV3MownGBGQDABKV1Q4dWI6CEKAOgsIZwnR6aC4WhguKFxA%2BfnVi%2B2pnrPFjah-ObPUUua42jRO5%2Bk9n66ekQMGD%2BOAAbT2gPEkic0xu1FoQJYbE4PD4-BhPQAukhpLsgA&css=Q&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwF8g)

**Comment by michealroberts** [comment]
@zyyv I'm going to open an MR for documentation on this ... it's not mentioned in either https://unocss.dev/config/theme#theme or https://unocss.dev/guide/config-file ... unless it is mentioned elsewhere?

**Comment by zyyv** [comment]
> @zyyv I'm going to open an MR for documentation on this ... it's not mentioned in either https://unocss.dev/config/theme#theme or https://unocss.dev/guide/config-file ... unless it is mentioned elsewhere?

Thanks, that will be help others to understand it.

**Comment by michealroberts** [comment]
@zyyv Thanks for your support on this, appreciated :) 🙏 Happy for this to be closed.

**Comment by zyyv** [comment]
> @zyyv Still doesn't seem to work, any ideas?
> 
> <img alt="Screenshot 2023-06-30 at 11 59 38" width="923" src="https://user-images.githubusercontent.com/84131395/250071877-db6d558a-df77-49ad-b6b5-da90ec63b8a4.png">

You have to follow the uno way to customize the animation.
[Playground](https://uno.antfu.me/play/?html=DwEwlgbgBAxgNgQwM5ILwHIAuBTAHpgWhmwDscAnKAWwICZ0A%2BKKAKGeAAdZEUMB3OgAYoCEmCoIcBPmADmsuNnRQARrIzlsIAgDZBgxjiSZgAeg4MWZ8BAZA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmcAvnOlBCHAOQCuWEAxgM6u0BQnqAHpLBQYAhvQA28NJhz5CwIgAoEnOHBgALVCFQAuRCtVxhWUMJjACe5YcMBrVAE9Kw7aysGbqgO7yiY3XAABggADACkADQAjCHhiDBQxqzo0CA6VDBmqAoAtADMaEQAlOQArHEICUkpUGlwGVkKBajFpKSBER6GpJ2eKPSJ5gRu%2Bn3evv56tFEcvZ49XarmINhEAGJYI9ZjPkR%2BAbSowqyoOdg5EPQwtHM2C33Ml1gwW4uGu-tT2NLAMKg3b3ungA9MC4B9JnQIag4DM4EcTmcsBcrnBvthfv9bmQ5gtSEVOEA&css=Q&options=N4IgzgLgTglgxhEAuAZgQwDZgKYBoTRoB2YKA9lALbLpbYC%2BQA)

**Comment by michealroberts** [comment]
Gah, this method @zyyv is horrible ... sorry but I am not happy with spreading every single nested facet of the theme inside each nested node ... there has to be a better way?

**Comment by michealroberts** [comment]
Yeh, this really is a terribly poor DX imho. Not to mention the previous issues with the lack of documentation.

"You have to follow the UnoCSS way" ...

Why diverge from Tailwind in such an unfriendly way! Gah.

**Comment by zyyv** [comment]
@michealroberts Yes, this DX is not friendly, but in order to synchronize other theme configurations, we rollup it together. If you have a better design method, please give us your opinions and suggestions.

BTW, There is a community preset https://github.com/unpreset/unocss-preset-useful, it optimizes the theme animate write style, hope you like it.
Refer to https://github.com/unpreset/unocss-preset-useful/blob/2750ad7ef72696c094e86c02ed21dfddd9c4a63d/test/utils.test.ts#L21-L36

**Comment by zyyv** [comment]
About https://github.com/unpreset/unocss-preset-useful, I will smooth out the differences between animation configurations in the next version. Provide a compatibility handle.
